```
Error running mint automation
+ mkdir -p 12186-4d7cf11/gopath/src/github.com/minio
+ git -C 12186-4d7cf11/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12186-4d7cf11/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12186-4d7cf11/gopath/src/github.com/minio/minio fetch --quiet minio pull/12186/head:pr12186
+ git -C 12186-4d7cf11/gopath/src/github.com/minio/minio checkout --quiet pr12186
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12186-4d7cf11/gopath
+ make -C 12186-4d7cf11/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
go: downloading cloud.google.com/go v0.57.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading google.golang.org/api v0.25.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/fatih/color v1.10.0
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/tinylib/msgp v1.1.3
go: downloading github.com/shirou/gopsutil/v3 v3.21.1
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading github.com/minio/argon2 v0.0.0-20210427164258-0025d10c2c04
go: downloading golang.org/x/crypto v0.0.0-20210415154028-4f45737414dc
go: downloading github.com/minio/sio v0.2.1
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/klauspost/compress v1.11.12
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/miekg/dns v1.1.35
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/pierrec/lz4 v2.5.2+incompatible
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/minio/csvparser v0.0.0-20210427163918-ee4f0ffa388d
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/tidwall/gjson v1.6.8
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/tidwall/pretty v1.0.2
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading go.uber.org/multierr v1.5.0
go: downloading go.uber.org/atomic v1.6.0
go: downloading cloud.google.com/go/storage v1.8.0
go: downloading google.golang.org/protobuf v1.24.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading google.golang.org/genproto v0.0.0-20200527145253-8367513e4ece
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading google.golang.org/grpc v1.29.1
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
# github.com/minio/minio/cmd
cmd/erasure-object.go:437:6: i declared but not used
make: *** [build] Error 2

```

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -433,6 +433,12 @@ func (er erasureObjects) getObjectFileInfo(ctx context.Context, bucket, object s
 		return fi, nil, nil, err
 	}
 
+	// if one of the disk is offline, return right here no need
+	// to attempt a heal on the object.
+	if countErrs(errs, errDiskNotFound) > 0 {
+		return fi, metaArr, onlineDisks, nil
+	}
+
 	var missingBlocks int
 	for i, err := range errs {
 		if err != nil && errors.Is(err, errFileNotFound) {

--- a/cmd/storage-rest-client.go
+++ b/cmd/storage-rest-client.go
@@ -238,6 +238,14 @@ func (client *storageRESTClient) SetDiskID(id string) {
 
 // DiskInfo - fetch disk information for a remote disk.
 func (client *storageRESTClient) DiskInfo(ctx context.Context) (info DiskInfo, err error) {
+	if !client.IsOnline() {
+		// make sure to check if the disk is offline, since the underlying
+		// value is cached we should attempt to invalidate it if such calls
+		// were attempted. This can lead to false success under certain conditions
+		// - this change attempts to avoid stale information if the underlying
+		// transport is already down.
+		return info, errDiskNotFound
+	}
 	client.diskInfoCache.Once.Do(func() {
 		client.diskInfoCache.TTL = time.Second
 		client.diskInfoCache.Update = func() (interface{}, error) {

